gateway-service/src/test/java/org/zowe/apiml/acceptance/X509SchemeTest.java

```diff
@@ -0,0 +1,123 @@
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.acceptance;
+
+import org.apache.http.Header;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.springframework.beans.factory.annotation.Value;
+import org.zowe.apiml.acceptance.common.AcceptanceTest;
+import org.zowe.apiml.acceptance.common.AcceptanceTestWithTwoServices;
+import org.zowe.apiml.acceptance.netflix.MetadataBuilder;
+import org.zowe.apiml.gateway.security.service.schema.X509Scheme;
+import org.zowe.apiml.util.config.SslContext;
+import org.zowe.apiml.util.config.SslContextConfigurer;
+
+import java.io.IOException;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.core.Is.is;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+/**
+ * This test verifies that only the client certificate is passed through the X509scheme to the southbound service.
+ * <p></p>
+ * Note: 2022/02/28 - current implementation of {@link X509Scheme} works with any certificate located in either default
+ * javax.servlet.request.X509Certificate attribute or in client.auth.X509Certificate custom attribute introduced
+ * by the {@link org.zowe.apiml.security.common.filter.ApimlX509Filter ApimlX509Filter}.
+ */
+@AcceptanceTest
+class X509SchemeTest extends AcceptanceTestWithTwoServices {
+
+    @Value("${server.ssl.keyStorePassword:password}")
+    private char[] keystorePassword;
+    @Value("${server.ssl.keyStore}")
+    private String keystore;
+    private final String clientKeystore = "../keystore/client_cert/client-certs.p12";
+
+
+    @BeforeEach
+    void setUp() throws Exception {
+        SslContextConfigurer configurer = new SslContextConfigurer(keystorePassword, clientKeystore, keystore);
+        SslContext.prepareSslAuthentication(configurer);
+
+        applicationRegistry.clearApplications();
+        MetadataBuilder customBuilder = MetadataBuilder.customInstance();
+        customBuilder.withX509();
+        MetadataBuilder defaultBuilder = MetadataBuilder.defaultInstance();
+        defaultBuilder.withX509();
+        applicationRegistry.addApplication(serviceWithDefaultConfiguration, defaultBuilder, false);
+        applicationRegistry.addApplication(serviceWithCustomConfiguration, customBuilder, true);
+        applicationRegistry.setCurrentApplication(serviceWithDefaultConfiguration.getId());
+
+        reset(mockClient);
+        mockValid200HttpResponse();
+    }
+
+    @Test
+    void givenValidClientCert_thenClientCertDetailsInRequestHeaders() throws IOException {
```

Comment:
Hey Petr, could you rather create Nested classes similarly to how is done in other tests? This is the convention we are trying to adopt, for a better readability of tests. An example is shown in `gateway-service/src/test/java/org/zowe/apiml/acceptance/RequestInstanceTest.java`

Reply:
Thanks Andrea, I'll keep that in mind. I fixed this test suite.